src/harness/fourslashImpl.ts

```diff
@@ -4510,6 +4510,58 @@ export class TestState {
 
         this.verifyCurrentFileContent(newFileContent);
     }
+
+    public baselineMapCode(
+        ranges: Range[],
+        changesFilename: string,
+    ): void {
+        const changes = this.getFileContent(changesFilename);
+        let fileName: string | undefined;
+        const focusLocations = ranges.map(({ fileName: fn, pos, end }) => {
+            if (!fileName) {
+                fileName = fn;
+            }
+            return [{ start: pos, length: end - pos }];
+        });
+        if (!fileName) {
+            throw new Error("No ranges passed in, or something went wrong.");
+        }
+        let before = this.getFileContent(fileName);
+        focusLocations.sort((a, b) => a[0].start - b[0].start);

```

Comment:
Is this requirement of API ? Does that need to be in the API jsdoc then?

Reply:
No. This is necessary to render the baselines properly (as in, figure out the right padding while writing all the `[|` and `|]` things out). It is irrelevant to the API.

Reply:
Then please move that for displaying and not for passing it to the API. API should be called with all the input parameters as is. The baselining code can do all sorts of sorting